src/redux/Books/books.js

```diff
@@ -0,0 +1,32 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+// action creators
+export const addBooks = (text) => ({
+  type: ADD_BOOK,
+  payload: {
+    text,
+  },
+});
+
+export const removeBooks = (text) => ({
+  type: REMOVE_BOOK,
+  payload: {
+    text,
+  },
+});
+
+// create reducers
+const initialState = [];
+
+const bookReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case ADD_BOOK:
+      return state.filter((books) => books.id !== payload);
```

Comment:
- [ ] Kindly make sure to add a new object in this case of `ADD_BOOK`. We only filter only when we are removing the book but by adding a new book please make sure you are adding a new object of the newly added book.

Reply:
thank you please a little clarification here am I supposed to use the map mehod in place of filter?

Reply:
Hello @Ngufor-emmanuella, When adding a new book we don't really need to iterate through our array, we only need to insert a new book object into our array containing all books. That action will not require iterating through the array.

Reply:
ok thanks